Docs: how to develop a webhook for a job integration (#3157)
kubernetes-sigs/kueue: site/content/en/docs/tasks/dev/integrate_a_custom_job.md

Add testing files for both the controller and the webhook.  You can check the test files in the other subfolders of `./pkg/controller/jobs/`
to learn how to implement them.

### Developing a mutation and validation webhook

Once you have implemented the `GenericJob` interface, you can register a webhook for the type by using
the `BaseWebhookFactory` in `pkg/controller/jobframework`. This webhook ensures that a job is created
in a suspended state and that queue names don't change when the job is admitted. You can read the
[MPIJob](https://github.com/kubernetes-sigs/kueue/tree/main/pkg/controller/jobs/mpijob) integration as an example.

> For certain use cases, you might need to add annotations or other fields or ensuring that certain features incompatible with Kueue are not used.
> In this case, you can register a `CustomDefaulter` using `WebhookManagedBy` in `pkg/controller/jobframework/webhook`
> and wrapping it with a `LosslessDefaulter`. You can read the [BatchJob](https://github.com/kubernetes-sigs/kueue/tree/main/pkg/controller/jobs/job)
> integration as an example.
> 
> The `LosslessDefaulter` ensures that the defaulter is compatible with future versions of the CRD,
> under the assumption that the defaulter does **not** remove any fields from the original object.
> If you need to restrict the usage of a field when using Kueue, you should consider rejecting the
> creation in the validator instead of dropping the field in the defaulter.

Reviewer: ```suggestion
For certain use cases, you might need to register a dedicated webhook. In this case, we suggest you use
the Kueue's `LosslessDefaulter`. which ensures that the defaulter is compatible with future versions of the
CRD, under the assumption that the defaulter does **not** remove any fields from the original object.

You can read the [BatchJob](https://github.com/kubernetes-sigs/kueue/tree/main/pkg/controller/jobs/job)
integration as an example.
```
This is my motivation for the suggestion:
-  "add annotations or other fields or ensuring that certain features incompatible with Kueue are not used" - I assume people registering a webhook will already know such basics.
- "you can register a `CustomDefaulter` using `WebhookManagedBy` in `pkg/controller/jobframework/webhook` and wrapping it with a `LosslessDefaulter`" - I assume this 
 can be learned quickly from the example by following the function calls
- "you should consider rejecting the creation in the validator instead of dropping the field in the defaulter." - I guess this is just one possibility, another would be to register another webhook for the type with the responsibility of dropping the field? But maybe we don't need to go too much into alternatives here?
- combined the paragraphs as they are small now, and moved the example to the bottom

Author: Done
